src/components/sound.js

```diff
@@ -125,24 +131,47 @@ module.exports.Component = registerComponent('sound', {
       evt.detail.cameraEl.getObject3D('camera').add(listener);
     });
 
-    sound = this.sound = new THREE.PositionalAudio(listener);
-    el.setObject3D(this.attrName, sound);
-
-    sound.source.onended = function () {
-      sound.onEnded();
-      el.emit('sound-ended');
-    };
+    this.soundsPoolGroup = new THREE.Group();
+    this.soundsPool = [];
+    // var soundGroup = new THREE.Group();
+    for (var i = 0; i < this.data.pool; i++) {
+      var sound = new THREE.PositionalAudio(listener);
+      this.soundsPool.push(sound);
+      this.soundsPoolGroup.add(sound);
+    }
+    el.setObject3D(this.attrName, this.soundsPoolGroup);
 
-    return sound;
+    this.soundsPool.forEach(function (sound) {
+      sound.source.onended = function () {
+        sound.onEnded();
+        el.emit('sound-ended', {index: i});
+      };
+    });
   },
 
   playSound: function () {
-    if (!this.sound.source.buffer) { return; }
-    this.sound.play();
+    for (var i = 0; i < this.soundsPool.length; i++) {
+      var sound = this.soundsPool[i];
+      if (!sound.isPlaying && sound.source.buffer) {
+        this.soundsPool[i].play();
+        return;
+      }
+    }
+    warn('Not unused sound found on pool, please increate its size using the `pool` attribute');
   },
 
   stopSound: function () {
-    if (!this.sound.source.buffer) { return; }
-    this.sound.stop();
+    this.soundsPool.forEach(function (sound) {
```

Comment:
I don't think so, as the user won't have the control of the playing buffer. For example if you have a gun you just keep shooting and suddenly you want to silent it but don't have any idea or way to know the exactly buffer that it's currently playing.
I believe we could leave it this way and if we get the need we could reevaluate it, as for example we'll need to give some feedback back to the entity to use it.